.github/workflows/rust-prs.yml

```diff
@@ -0,0 +1,459 @@
+        shell: bash
+        run: |
+          cargo nextest run --target ${{ matrix.platform.target }} --no-run --profile ci ${{ fromJson(needs.what-changed.outputs.rust).cargo-test-specs }}
+
+      - name: Run tests (non docker platforms)
+        id: tests
+        if: fromJson(needs.what-changed.outputs.rust).cargo-test-specs && runner.os != 'Linux' && matrix.platform.target != 'x86_64-apple-darwin'
+        shell: bash
+        run: |
+          cargo nextest run --target ${{ matrix.platform.target }} --profile ci ${{ fromJson(needs.what-changed.outputs.rust).cargo-test-specs }}
+
+      - name: Build tests (docker platforms)
+        if: fromJson(needs.what-changed.outputs.rust).cargo-docker-test-specs && runner.os == 'Linux'
+        shell: bash
+        run: |
+          cargo nextest run --target ${{ matrix.platform.target }} --no-run --profile ci ${{ fromJson(needs.what-changed.outputs.rust).cargo-test-specs }}
+
+      - name: Run tests (docker platforms)
+        id: tests_docker
+        if: fromJson(needs.what-changed.outputs.rust).cargo-docker-test-specs && runner.os == 'Linux'
+        shell: bash
+        run: |
+          cargo nextest run --target ${{ matrix.platform.target }} --profile ci ${{ fromJson(needs.what-changed.outputs.rust).cargo-test-specs }}
+
+      - name: Upload the JUnit files
+        if: ${{ ( success() || failure() ) && !contains(steps.tests.outputs.exitcode, '101') && !contains(steps.tests_docker.outputs.exitcode, '101') && matrix.platform.target != 'x86_64-apple-darwin'}}
+        uses: ./.github/actions/test_upload_datadog
+        with:
+          api_key: ${{ secrets.DATADOG_API_KEY }}
+          junit_path: target/nextest/ci/junit.xml
+          service: cli
+
+      - name: Build debug binaries
+        if: fromJson(needs.what-changed.outputs.rust).cargo-bin-specs
+        shell: bash
+        run: |
+          cargo build --target ${{ matrix.platform.target }} ${{ fromJson(needs.what-changed.outputs.rust).cargo-bin-specs }}
+
+  docker-gateway:
+    needs: [what-changed]
+    if: |
+      needs.what-changed.outputs.gateway-docker == 'true'
+      || contains(fromJson(needs.what-changed.outputs.rust).changed-packages, 'grafbase-gateway')
+    runs-on: ubicloud-standard-16
+    permissions:
+      packages: write
+    env:
+      COMMIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+    steps:
+      - name: Get sources
+        uses: actions/checkout@v4
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build docker images
+        run: |
+          # Re-use the latest layers if possible
+          docker pull ghcr.io/grafbase/gateway:latest || true
+          docker build -f gateway/Dockerfile -t ghcr.io/grafbase/gateway:$COMMIT_SHA .
```

Comment:
nice!

Reply:
This was lifted straight out of the existing workflow fwiw